pkg/deploy/apiserver/apiserver.go

```diff
@@ -102,7 +103,11 @@ func (c *AppsConfig) newV1RESTStorage() (map[string]rest.Storage, error) {
 	// TODO sort out who is using this and why.  it was hardcoded before the migration and I suspect that it is being used
 	// to serialize out objects into annotations.
 	externalVersionCodec := kapi.Codecs.LegacyCodec(schema.GroupVersion{Group: "", Version: "v1"})
-	deprecatedOpenshiftClient, err := osclient.New(c.GenericConfig.LoopbackClientConfig)
+	openshiftInternalAppsClient, err := appsclientinternal.NewForConfig(c.CoreAPIServerClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	openshiftInternalImageClient, err := imageclientinternal.NewForConfig(c.CoreAPIServerClientConfig)
```

Comment:
this one is using the correct clientconfig, since the apps server doesn't host images.

Reply:
We should probably write it down for others to be aware of that rule, for the future time where we'll split each group into its own apiserver.